lib/ansible/modules/cloud/azure/azure_rm_aci.py

```diff
@@ -0,0 +1,503 @@
+            self.log("Container Group doesn't exist")
+
+            if self.state == 'absent':
+                self.log("Nothing to delete")
+            else:
+                to_be_updated = True
+        else:
+            self.log("Container instance already exists")
+
+            if self.state == 'absent':
+                self.delete_aci()
+                self.results['changed'] = True
+                self.log("ACI instance deleted")
+            elif self.state == 'present':
+                self.log("Need to check if container group has to be deleted or may be updated")
+                to_be_updated = self.check_need_update(response)
+                if to_be_updated:
+                    self.log('Deleting ACI instance before update')
+                    self.delete_aci()
+
+        if self.state == 'present':
+
+            self.log("Need to Create / Update the ACI instance")
+
+            if self.check_mode:
+                return self.results
+
+            if to_be_updated:
+                self.results['state'] = self.create_update_aci()
+                self.results['changed'] = True
+            else:
+                self.results['state'] = response
+
+            self.log("Creation / Update done")
+
+        return self.results
+
+    def create_update_aci(self):
+        '''
+        Creates or updates a container service with the specified configuration of orchestrator, masters, and agents.
+
+        :return: deserialized ACI instance state dictionary
+        '''
+        self.log("Creating / Updating the ACI instance {0}".format(self.name))
+
+        credentials = None
+
+        if self.registry_login_server is not None:
+            credentials = ImageRegistryCredential(self.registry_login_server,
+                                                  self.registry_username,
+                                                  self.registry_password)
+
+        ip_address = None
+
+        if self.ip_address is not None:
+            # get list of ports
+            if self.ports:
+                ports = []
+                for port in self.ports:
+                    ports.append(Port(port, "TCP"))
+                ip_address = IpAddress(ports, self.ip_address)
+
+        containers = []
+
+        for container_def in self.containers:
+            name = container_def.get("name")
+            image = container_def.get("image")
+            memory = container_def.get("memory")
+            cpu = container_def.get("cpu")
+            ports = []
+
+            port_list = container_def.get("ports")
+            if port_list:
+                for port in port_list:
+                    ports.append(ContainerPort(port))
+
+            if cpu is None:
+                cpu = 1
+            if memory is None:
+                memory = 1.5
+            containers.append(Container(name, image, ResourceRequirements(ResourceRequests(memory, cpu)), None, ports))
+
+        parameters = ContainerGroup(self.location,
```

Comment:
pls add a comment to show the parameter name of `ContainerGroup()`

Reply:
Using named arguments will help readability